### DISCUSSION TITLE
Setting Up and Testing Email Expiring Notifications

### DISCUSSION THREAD

**Discussion by aleng1** [Keyfactor/ejbca-ce#453] [Q&A] [answered]
Hello,

I'm new to EJBCA and would like to set up email notifications for certificate expiration. While I'm interested in configuring this feature, I also want to make sure it works correctly by testing the email notifications before the certificates actually expire.

Could someone please provide guidance on:

1) How to set up and configure email notifications for certificate expiration?
2) How to test the email notifications to ensure they are being sent to the correct email addresses and with the correct content?
I'd appreciate any step-by-step instructions, recommended configurations, and best practices for implementing and testing this feature.

Thank you for your help!

**Comment by aleng1** [comment]
anyone can answer to this?🙏

**Reply by merryo11** [reply]
You can create service for expiry notification using admin GUI. It is located at System functions menu > Services.

https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/services

**Reply by aleng1** [reply]
Yea, I tried creating a new service using the Certificate Expiration Checker as the worker, but I haven't received any emails. Do I need to configure anything, such as SMTP services or other settings?

**Reply by merryo11** [reply, marked as answer]
If you are using EJBCA with docker then yes there are environment variables like SMTP_DESTINATION, SMTP_FROM, SMTP_USERNAME & SMTP_PASSWORD which you can use to setup your smtp settings.

**Reply by merryo11** [reply]
use them while creating your docker container. refer to https://hub.docker.com/r/keyfactor/ejbca-ce link under Sending Email heading

**Reply by aleng1** [reply]
version: '3'
networks:
  access-bridge:
    driver: bridge
  application-bridge:
    driver: bridge
services:
  ejbca-database:
    container_name: ejbca-database
    image: "library/mariadb:latest"
    networks:
      - application-bridge
    environment:
      - MYSQL_ROOT_PASSWORD=foo123
      - MYSQL_DATABASE=ejbca
      - MYSQL_USER=ejbca
      - MYSQL_PASSWORD=ejbca
    volumes:
      - ./datadbdir:/var/lib/mysql:rw
  ejbca-node1:
    hostname: ejbca-node1
    container_name: ejbca
    image: keyfactor/ejbca-ce:latest
    depends_on:
      - ejbca-database
    networks:
      - access-bridge
      - application-bridge
    environment:
      - DATABASE_JDBC_URL=jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8
      - LOG_LEVEL_APP=INFO
      - LOG_LEVEL_SERVER=INFO
      - TLS_SETUP_ENABLED=simple
      - SMTP_DESTINATION=your_smtp_server
      - SMTP_FROM=your_email_address
      - SMTP_USERNAME=your_smtp_username
      - SMTP_PASSWORD=your_smtp_password
    ports:
      - "80:8080"
      - "443:8443"


Is this correct configuration of SMTP for .yaml file?

**Reply by merryo11** [reply]
Yes seems like good. After container use docker inspect command to see what other SMTP variables are available for you if the above doesn't work. The above .yaml file shall establish the connection and service created through admin gui shall help you further.
